File: ReactNativeApp/android/app/src/main/java/com/ssvc/Controller.java
package com.ssvc;

import com.render.DebugText;

abstract public class Controller<S extends State> implements StateObserver, DebugText {
    protected S state;

    public Controller(S state) {
        state.addObserver(this);
        this.state = state;
    }

    @Override
    public void onNotify() {

    }
}

File: ReactNativeApp/ios/Modules/SSVC/Base/Base/Controller.swift
//
//  Controller.swift
//  jogo
//
//  Created by Mohsin on 12/10/2021.
//

import Foundation

public class Controller: ControllerType, StateObserverType, CleanupProtocol {
  internal var state: State?
  public var type: ObserverType = .Controller
  
  init(state: State) {
    self.state = state
    self.state?.addObserver(stateObserver: self)
  }
  
  public func onNotify() {}
  
  public func removeSVs(){}
  
  public func cleanup() {
    self.state = nil
  }
  
}


// MARK: Debug Text Extension
extension Controller: DebugText {
  @objc public func getDebugText() -> String { return "" }
}
